spec/requests/generic_object_definitions_spec.rb

```diff
@@ -397,15 +392,14 @@
     end
 
     it 'can remove attributes by id, name, or href' do
-      object_def.add_property_attribute('attr1', 'string')
       object_def2.add_property_attribute('attr2', 'string')
       object_def3.add_property_attribute('attr3', 'string')
       api_basic_authorize collection_action_identifier(:generic_object_definitions, :add_associations)
 
       request = {
         'action'    => 'remove_attributes',
         'resources' => [
-          { 'name' => object_def.name, 'resource' => { 'attributes' => { 'attr1' => 'string' } } },
+          { 'name' => object_def.name, 'resource' => { 'attributes' => { 'widget' => 'string' } } },
```

Comment:
I think this test might now be less clear because it's no longer obvious what the connection is between this attribute and `object_def` - one of the main players in this test. It's now implicit that it has this specific attribute. I think traits are fine if they help you create objects that are suited for certain use cases, but where the specifics don't matter too much ("an object that has associations, attributes, etc..."). But when those specifics are the thing under test, I think those details are often best left in the test body, even if that means duplicating things from the trait definition.

Reply:
@imtayadeway I agree, was actually working on these changes.